tests/test_home_meter.py

```diff
@@ -0,0 +1,242 @@
+        self.strategy.state.delete_past_state_values = Mock()
+        self.strategy.bid_update.delete_past_state_values = Mock()
+        self.strategy.offer_update.delete_past_state_values = Mock()
+        self.strategy._set_energy_forecast_for_future_markets = Mock()
+        self.strategy._post_offer = Mock()
+        market_mocks = self._create_market_mocks(3)
+        self.strategy.area.all_markets = market_mocks
+
+        self.strategy.event_market_cycle()
+        super_method_mock.assert_called_once()
+        self.strategy.bid_update.update_and_populate_price_settings.assert_called_with(
+            self.area_mock)
+        self.strategy.offer_update.update_and_populate_price_settings.assert_called_with(
+            self.area_mock)
+        self.strategy.bid_update.reset.assert_called_with(self.strategy)
+        self.strategy.offer_update.reset.assert_called_with(self.strategy)
+
+        # We just assert calls to private methods, because we have separate unittests for them
+        self.strategy._set_energy_forecast_for_future_markets.assert_called_once_with(
+            reconfigure=False)
+        assert self.strategy._post_offer.call_count == 3
+
+        self.strategy.state.delete_past_state_values.assert_called_once_with(
+            self.area_mock.current_market.time_slot)
+        self.strategy.bid_update.delete_past_state_values.assert_called_once_with(
+            self.area_mock.current_market.time_slot)
+        self.strategy.offer_update.delete_past_state_values.assert_called_once_with(
+            self.area_mock.current_market.time_slot)
+
+    @patch("d3a_interface.constants_limits.ConstSettings.IAASettings")
+    def test_event_offer_two_sided_market(self, iaa_settings_mock):
+        """The device does not automatically react to offers in two-sided markets."""
+        iaa_settings_mock.MARKET_TYPE = 2
+        self.strategy.event_offer(market_id="some_market_id", offer="some_offer")
+        # Make sure that the method is returning immediately and not executing further logic
+        self.strategy.area.get_future_market_from_id.assert_not_called()
+
+    def test_post_offer(self):
+        """Offers are generated and sent to the offers class to be posted.
+
+        This method is private, but we test it to avoid duplication and because of its complexity.
+        """
+        self.strategy.state.get_available_energy_kWh = Mock(return_value=12)
+        market_mock = self._create_market_mocks(1)[0]
+        offer_mock = Mock()
+        market_mock.offer = Mock(return_value=offer_mock)
+        self.strategy.offer_update.initial_rate = {market_mock.time_slot: 11}
+        self.strategy.offers.post = Mock()
+
+        self.strategy._post_offer(market_mock)
+        self.strategy.offers.post.assert_called_once_with(offer_mock, market_mock.id)
+
+    @staticmethod
+    def _create_profile_mock():
+        # NOTE: the datetimes should be the same that will be used by the market slots
```

Comment:
Sure. The energy profile must be filled with data that will be used by the market slots. If we create the market slots (mocks) using different timestamps, they will not use the energy readings provided in the profile.
Do you suggest rephrasing the comment somehow, or moving it elsewhere maybe?

Reply:
Ok, I understand now. I was confused because that should be of course the case.
It is really minor and feel free to skip it, but if you want to avoid this comment, you could expose the `slot_times` list in self and re-use it in `_create_market_mocks` with this you can be sure to create the fitting markets. Again, feel free to skip it, it is just the test.